src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -205,9 +204,17 @@ private float addRow(
       RowBufferStats forkedStats = statsMap.get(columnName).forkEmpty();
       forkedStatsMap.put(columnName, forkedStats);
       ColumnMetadata column = parquetColumn.columnMetadata;
-      ParquetValueParser.ParquetBufferValue valueWithSize =
-          ParquetValueParser.parseColumnValueToParquet(
-              value, column, parquetColumn.type, forkedStats, defaultTimezone, insertRowsCurrIndex);
+      ParquetBufferValue valueWithSize =
+          (column.getSourceIcebergDataType() == null
+              ? SnowflakeParquetValueParser.parseColumnValueToParquet(
+                  value,
+                  column,
+                  parquetColumn.type.asPrimitiveType().getPrimitiveTypeName(),
+                  forkedStats,
+                  defaultTimezone,
+                  insertRowsCurrIndex)
+              : IcebergParquetValueParser.parseColumnValueToParquet(
+                  value, parquetColumn.type, forkedStats, defaultTimezone, insertRowsCurrIndex));
```

Comment:
no need to pass in columnMetadata / columnMetadata.sourceIcebergDataType?

Reply:
I didn't use the column metadata as `parquetColumn.type` already include all information we need including precision, scale, length, etc.